src/zenml/integrations/argilla/annotators/argilla_annotator.py

```diff
@@ -0,0 +1,299 @@
+#  Copyright (c) ZenML GmbH 2024. All Rights Reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at:
+#
+#       https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+"""Implementation of the Argilla annotation integration."""
+
+import json
+from typing import Any, List, Tuple, Type, cast
+
+import argilla as rg
+from argilla.client.client import Argilla as ArgillaClient
+from argilla.client.sdk.commons.errors import BaseClientError, NotFoundApiError
+
+from zenml.annotators.base_annotator import BaseAnnotator
+from zenml.integrations.argilla.flavors import (
+    ArgillaAnnotatorSettings,
+)
+from zenml.integrations.argilla.flavors.argilla_annotator_flavor import (
+    ArgillaAnnotatorConfig,
+)
+from zenml.logger import get_logger
+from zenml.stack.authentication_mixin import AuthenticationMixin
+
+logger = get_logger(__name__)
+
+
+class ArgillaAnnotator(BaseAnnotator, AuthenticationMixin):
+    """Class to interact with the Argilla annotation interface."""
+
+    @property
+    def config(self) -> ArgillaAnnotatorConfig:
+        """Returns the `ArgillaAnnotatorConfig` config.
+
+        Returns:
+            The configuration.
+        """
+        return cast(ArgillaAnnotatorConfig, self._config)
+
+    @property
+    def settings_class(self) -> Type[ArgillaAnnotatorSettings]:
+        """Settings class for the Argilla annotator.
+
+        Returns:
+            The settings class.
+        """
+        return ArgillaAnnotatorSettings
+
+    def get_url(self) -> str:
+        """Gets the top-level URL of the annotation interface.
+
+        Returns:
+            The URL of the annotation interface.
+        """
+        return (
+            f"{self.config.instance_url}:{self.config.port}"
+            if self.config.port
+            else self.config.instance_url
+        )
+
+    def _get_client(self) -> ArgillaClient:
+        """Gets Argilla client.
+
+        Returns:
+            Argilla client.
+
+        Raises:
+            ValueError: when unable to access the Argilla API key.
+        """
+        config = self.config
+        if config.api_key and config.authentication_secret:
+            raise ValueError(
+                "Both API key and authentication secret are provided. "
+                "Please provide only one."
+            )
```

Comment:
why not prioritize only one of them even if they are both passed, also we don't check if non of them are passed

Reply:
Good point. I updated to address this.